File: Java/src/main/java/myservice/MyService.java
package myservice;

import sso.Request;
import sso.Response;
import sso.SingleSignOnRegistry;

public class MyService {
    
    private SingleSignOnRegistry registry;
    
    public MyService(SingleSignOnRegistry registry) {
        this.registry = registry;
    }
    
    public Response handleRequest(Request request) {
        if (!registry.is_valid(request.getSSOToken())) {
            return new Response("Please sign in");
        }

        return new Response("hello " + request.getName() + "!");
    }
}

File: php/src/MyService/MyService.php
<?php

namespace MyService;

use SSO\Request;
use SSO\Response;
use SSO\SingleSignOnRegistry;

class MyService
{
    /**
     * @var SingleSignOnRegistry
     */
    private $registry;

    /**
     * @param SingleSignOnRegistry $registry
     */
    public function __construct(SingleSignOnRegistry $registry)
    {
        $this->registry = $registry;
    }

    /**
     * @param Request $request
     * @return Response
     */
    public function handleRequest(Request $request)
    {
        return new Response("hello " . $request->getName() . "!");
    }
}
